src/libraries/Common/src/System/Net/Http/aspnetcore/Http2/Hpack/Huffman.cs

```diff
@@ -307,128 +402,143 @@ public static (uint encoded, int bitLength) Encode(int data)
         /// <returns>The number of decoded symbols.</returns>
         public static int Decode(ReadOnlySpan<byte> src, ref byte[] dstArray)
         {
+            // The code below implements the decoding logic for an HPack huffman encoded literal values.
+            // https://httpwg.org/specs/rfc7541.html#string.literal.representation
+            //
+            // To decode a symbol, we traverse the decoding lookup table tree by 8 bits for each lookup
+            // until we found a leaf - which contains decoded symbol (octet)
+            //
+            // see comments in GenerateDecodingLookupTree() describing decoding table
+
             Span<byte> dst = dstArray;
             Debug.Assert(dst != null && dst.Length > 0);
 
+            if (s_decodingTree == null)
+            {
+                Volatile.Write(ref s_decodingTree, GenerateDecodingLookupTree());
+            }
+            ushort[] decodingTree = s_decodingTree;
```

Comment:
Technically this should be more like:
```suggestion
            ushort[] decodingTree = Volatile.Read(ref s_decodingTree);
            if (decodingTree is null)
            {
                decodingTree = GenerateDecodingLookupTree();
                Volatile.Write(ref s_decodingTree, decodingTree);
            }
```

That said, why is this not just:
```C#
private static readonly ushort[]? s_decodingTree = GenerateDecodingLookupTree();
```
?  Isn't it the case that if this type is being used, this s_decodingTree is going to be needed?

Reply:
This is good point, In future, this class might handle encoding as well, but even in such cases, the situation that only encoding is used and decoding is not will be probably rare. As GenerateDecodingLookupTree is just ~5us we might not overoptimize it here. I will change it to `private static readonly ushort[] s_decodingTree = GenerateDecodingLookupTree();`